akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -945,6 +945,13 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
         val newGossip = localGossip.markAsDown(m)
         updateLatestGossip(newGossip)
         publishMembershipState()
+        if (address == cluster.selfAddress) {
+          // spread the word quickly, without waiting for next gossip tick
+          gossipRandomN(MaxGossipsBeforeShuttingDownMyself)
+        } else {
+          // try to gossip immediately to downed node, as a STONITH signal
+          gossipTo(m.uniqueAddress)
```

Comment:
This is what makes biggest difference. Saves around 4 seconds in the tests I tried (downing 12 of 26 nodes).

Reply:
Nice